crates/pathfinder/src/state/sync/l1.rs

```diff
@@ -0,0 +1,104 @@
+use pathfinder_ethereum::{L1StateUpdate, StarknetEthereumClient};
+use tokio::sync::mpsc::Sender;
+
+/// Syncs L1 state updates.
+pub async fn sync(
+    tx_event: Sender<L1StateUpdate>,
+    ethereum_client: StarknetEthereumClient,
+    poll_interval: std::time::Duration,
+) {
+    let mut backoff = RetryBackoff::new(std::time::Duration::from_secs(1), poll_interval);
+    loop {
+        tokio::time::sleep(backoff.delay()).await;
+
+        match ethereum_client.get_starknet_state().await {
+            Ok(state) => {
+                backoff.success();
+                if let Err(e) = tx_event.send(state).await {
+                    tracing::error!(reason=?e, "L1 event sending failed");
+                }
+            }
+            Err(e) => {
+                backoff.failure();
+                tracing::error!(reason=?e, "L1 call failed");
+            }
+        }
+    }
+}
+
+struct RetryBackoff {
+    ok: bool,
+    min_millis: u32,
+    max_millis: u32,
+    delay_millis: u32,
+}
```

Comment:
I think we have a more complex version of this somewhere already -- do you specifically want this simpler form?

Reply:
Why use "complex version" when this one works perfectly? :)

The enforced retry of a periodic action does not make much sense IMHO, we can just update the delay on failure.

Reply:
The "complex version" can be easily restored if needed. I didn't find any other usages but here.